### DISCUSSION TITLE
Cline Feature Requests

### DISCUSSION THREAD

**Discussion by saoudrizwan** [cline/cline#373] [Announcements]
We’re using Discussions as a place for the community to vote on their favorite feature requests!

[Join the Discord](https://discord.gg/cline) for announcements, sharing projects, talking with friends and more!

**Comment by RaySinner** [comment]
I would be glad to see more settings so that we can try to configure work with backward models, for example DeepSeek 2.5, which may not have developed “tools”; it writes code objectively better than others.

**Comment by TomLucidor** [comment]
Game, but Discord is kinda iffy, dumping things here feels slightly easier

**Comment by mehranmahouti** [comment]
Can we add agent logic to the program? one agent can check the code, another implement it, deploy it or check it for more advanced properties.

**Reply by rmounikkumar** [reply]
I have a referral link. AgentRouter is currently offering a $50 new-user bonus, and the referral campaign says the referrer receives $150. Check the campaign terms before signing up.


https://agentrouter.org/register?aff=yB1n

**Comment by Huge** [comment]
parallel processing through multiple models + majority voting or answer/proposal-merging model

**Reply by TomLucidor** [reply]
But then we would need something closer to OpenRouter and not Ollama, cus that will be computationally expensive

**Comment by snowkings** [comment]
I notice that the cost per prompt for G 2.5 doesn't appear in the task chat parameters box and perhaps this is being worked on but here is an LLM calculator from Simon Wilison. 

https://tools.simonwillison.net/llm-prices?utm_source=substack&utm_medium=email

**Comment by brandonoddo** [comment]
Feature Request - Could Cline access Cline task history? Example: Scraping lessons learned for a given project type.

**Reply by TomLucidor** [reply]
Seconding this but for something like RooFlow, self-improving agents shouldn't be gatekept

**Comment by RobertBernstein** [comment]
Add an option to enable the user to specify a different (free?) model from plan and/or act for writing commit comments.

**Comment by nothingbutgames1967-LKU** [comment]
Allow an option to stop autoscroll when generating please.

**Comment by dougame841114** [comment]
Adding my 2 cents on API cost optimization since that's a big pain point for Cline users.

If you're tired of paying premium prices for every coding request, check out JZS Token (https://jzstoken.com). It's a multi-model API gateway with 40+ models behind one OpenAI-compatible endpoint. Cline works out of the box: just set the provider to OpenAI Compatible, base URL to https://jzstoken.com/api/v1, and you're good to go.

Here's their pricing (USD per 1M tokens) for some popular models:

| Model | Input | Output | Cache Read |
|-------|-------|--------|------------|
| jzs-max-3.0 | $0.0620 | $0.3100 | $0.0062 |
| LongCat-2.0 | $0.0775 | $0.3100 | $0.0077 |
| MiniMax-M3 | $0.0775 | $0.3875 | $0.0077 |
| MiMo v2.5 | $0.1550 | $0.7750 | $0.0155 |
| gpt-5.6-terra | $0.2325 | $1.39 | $0.0232 |
| gpt-5.5 | $0.3875 | $2.32 | $0.0387 |
| claude-opus-5 | $1.16 | $5.81 | $0.1162 |

For Cline specifically: use the cheap models for routine refactoring and simple tasks, switch to premium models only when you need heavy reasoning. This alone cuts costs by 60-70%.

The gateway also supports all three protocols (Chat Completions, Responses, Messages), so any Cline feature just works. No fiddling with configs.

Hope this helps someone! Cline is great, but the API costs add up fast. Having a cheaper option helps.
